jupyterhub_traefik_proxy/proxy.py

```diff
@@ -557,6 +589,8 @@ async def add_route(self, routespec, target, data):
         The proxy implementation should also have a way to associate the fact that a
         route came from JupyterHub.
         """
+        if self._start_future and not self._start_future.done():
+            await self._start_future
```

Comment:
we're not also awaiting `_start_external`  when a route is deleted because we're assuming this operation will never happen first?

Reply:
Yeah, it's not likely to make sense to delete a route first since you can't know what routes exist yet. JupyterHub will never do this.